UT4MasterServer/Controllers/PersonaController.cs

```diff
@@ -32,58 +33,91 @@ public async Task<IActionResult> AccountLookup([FromQuery(Name = "q")] string qu
+			{ "displayName", account.Username },
+			{ "extenalAuths", new JObject() }
+		};
 
 		return Json(obj);
 	}
 
+	[HttpPost("accounts/search")]
+	public async Task<IActionResult> SearchAccounts([FromBody] AccountSearchRequest request)
+	{
+		if (User.Identity is not EpicUserIdentity authenticatedUser)
+		{
+			return Unauthorized();
+		}
+		var accounts = await accountService.SearchAccountsAsync(request.Query);
+
+		if (request.Roles != null && request.Roles.Length > 0)
+		{
+			accounts = accounts.Where((a) => a.Roles != null && a.Roles.Intersect(request.Roles).Any());
+		}
+
+		var count = accounts.Count();
+
+		accounts = accounts.Skip(request.Skip).Take(request.Take);
+
+		if (request.IncludeRoles)
+		{
+			return Ok(
+				new
+				{
+					accounts = accounts.Select((account) => new { account.ID, account.Username, account.Roles }),
+					count
+				}
+			);
+		}
+		return Ok(
+				new
+				{
+					accounts = accounts.Select((account) => new { account.ID, account.Username, }),
+					count
+				}
+			);
+	}
+
 	[HttpGet("account/{id}")]
 	public async Task<IActionResult> GetAccount(string id)
 	{
 		if (User.Identity is not EpicUserIdentity authenticatedUser)
-			return Unauthorized();
-
-		EpicID eid = EpicID.FromString(id);
+		{
+			return Unauthorized();
+		}
+
+		var eid = EpicID.FromString(id);
 
 		if (eid != authenticatedUser.Session.AccountID)
+		{
 			return Unauthorized();
+		}
 
 		logger.LogInformation($"{authenticatedUser.Session.AccountID} is looking for account {id}");
 
 		var account = await accountService.GetAccountAsync(eid);
 		if (account == null)
+		{
 			return NotFound();
+		}
 
-		return Json(account);
-	}
-
-	[HttpGet("accounts")]
-	public async Task<IActionResult> GetAllAccounts()
-	{
-		if (User.Identity is not EpicUserIdentity authenticatedUser)
-			return Unauthorized();
-
-		var accounts = await accountService.GetAllAccountsAsync();
-		logger.LogInformation($"{authenticatedUser.Session.AccountID} is looking for all accounts");
-
-		// create json response
-		var arr = new JArray();
-		// TODO: Limit to 1000 for now, just to not allow unlimited access to the accounts collection. Should be replaced with paging or a search function at some point.
-		foreach (var account in accounts.Take(1000))
-		{
-			var obj = new JObject();
-			obj.Add("ID", account.ID.ToString());
-			obj.Add("Username", account.Username);
-			var roles = new JArray();
-			roles.Add(account.Roles);
-			obj.Add("Roles", roles);
-			arr.Add(obj);
-		}
-
-		return Json(arr);
-	}
+		return Ok(account);
+	}
 
+	[HttpPost("accounts")]
+	public async Task<IActionResult> GetAccountsByIds([FromBody] string[] ids)
+	{
+		if (User.Identity is not EpicUserIdentity authenticatedUser)
+		{
+			return Unauthorized();
+		}
+
+		var eIds = ids.Distinct().Select(x => EpicID.FromString(x));
+		var accounts = await accountService.GetAccountsAsync(eIds);
+		logger.LogInformation($"{authenticatedUser.Session.AccountID} is looking for limited accounts by ID");
```

Comment:
Don't use string interpolation in logging, use this:
```charp
logger.LogInformation("{AccountId} is looking for limited accounts by ID", authenticatedUser.Session.AccountID);
```

Reply:
I just copied this from another function, so i don't mind changing it, but curious how that syntax would even work? Are we parsing the strings going into the logger or something?

Reply:
When logs are exported to something like Kibana/ElasticSearch, {AccountId} becomes a variable that you can use to search by.

Reply:
So is it a specific set of variables that we have access to in the context of a logged method invocation? Like if I wanted to log something specific to that occurrence, say the result of a calculation or something that only that invocation would have access to, I'd still need to interpolate that, right?

Reply:
No, the variable is whatever you put in {}. Once the logs are dumped to a service like ElasticSearch, the service scans all logs and extracts all variables so you can use them in filters. When using these variables, you should be consistent with naming and data type. If you log once a string as {AccountId}, you should stick with it, and don't change it to int or something else.

Basic data types are supported like bools, numbers, strings and dates. For example, for dates you could use filter between or greaten than, etc.

Reply:
Oh, duh, I didn't see that you were passing the ID as a second param, that's why I was confused. That's what I get for looking at comments on my phone.